src/storage.h

```diff
@@ -206,28 +228,177 @@ class Storage
                       "uilabel":"Batch Handling"}
   bool discrete_handling;
 
+  #pragma cyclus var {"default": "Fixed",\
+                      "tooltip": "Type of active buying frequency",\
+                      "doc": "Options: Fixed, Uniform, Normal. Fixed requires active_buying_val. Uniform "\
+                      "requires active_buying_max, with optional active_buying_min (default 1). Normal "\
+                      "requires active_buying_mean and active_buying_std, with optional "\
+                      "active_buying_min and active_buying_max.",\
+                      "uitype": "combobox",\
+                      "categorical": ["Fixed", "Uniform", "Normal"],\
+                      "uilabel": "Active Buying Frequency Type"}
+  std::string active_buying_frequency_type;
+
+  #pragma cyclus var {"default": 1,\
+                      "tooltip": "Fixed active buying frequency",\
+                      "doc": "The length in time steps of the active buying period. Required for fixed "\
+                      "active_buying_frequency_type. Must be greater than or equal to 1 (i.e., agent "\
+                      "cannot always be dormant)",\
+                      "uitype": "range", \
+                      "range": [1, 1e299], \
+                      "uilabel": "Active Buying Frequency Value"}
+  int active_buying_val;
+
+  #pragma cyclus var {"default": 1,\
+                      "tooltip": "Active buying distribution minimum",\
+                      "doc": "The minimum length in time steps of the active buying period. Optional for "\
+                      "Uniform and Normal active_buying_frequency_type. Must be greater than or equal to 1 ",\
+                      "uitype": "range", \
+                      "range": [1, 1e299], \
+                      "uilabel": "Active Buying Frequency Minimum"}
+  int active_buying_min;
+
+  #pragma cyclus var {"default": 1e299,\
```

Comment:
Because you define a default, XML won't complain if this is missing from the input file, and it will always be set with a value.  This means that it's technically optional since we have no way to recognize that it's missing.  Normally, if we require an input, we don't set a default, but since you have different requirements for different combinations, we can't really do that here.

One option would be to set the default as something invalid (e.g. -1) and then add a check for that value as an indication that there was nothing in the input file.

Another option is that we tell users that if they don't set it, they get 1e299 for that value, which could be pretty pathological 😀

Reply:
The same goes for some of the other entries here.